ci-operator/step-registry/mirror-images/icsp-olm-operators/mirror-images-icsp-olm-operators-commands.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+mirror_auths="${SHARED_DIR}/mirror_auths"
+
+# private mirror registry host
+# <public_dns>:<port>
+MIRROR_REGISTRY_HOST=`head -n 1 "${SHARED_DIR}/mirror_registry_url"`
+if [ ! -f "${SHARED_DIR}/mirror_registry_url" ]; then
+    echo "File ${SHARED_DIR}/mirror_registry_url does not exist."
+    exit 1
+fi
+echo "MIRROR_REGISTRY_HOST: $MIRROR_REGISTRY_HOST"
+
+# add the custom registry auth to the .dockerconfigjson of the cluster
+registry_cred=`head -n 1 "/var/run/vault/mirror-registry/registry_creds" | base64 -w 0`
+jq --argjson a "{\"${MIRROR_REGISTRY_HOST}\": {\"auth\": \"$registry_cred\"}}" '.auths |= . + $a' "/var/run/vault/mirror-registry/dockerconfigjson" > "${mirror_auths}"
+
+# Don't mirror OLM operators images, but create the ICSP for them.
+echo "=============>>> create ICSP for OLM operators"
+echo "$ oc adm catalog mirror -a ${mirror_auths} quay.io/openshift-qe-optional-operators/ocp4-index:latest ${MIRROR_REGISTRY_HOST} --manifests-only --to-manifests=/tmp/olm_mirror"
+if oc adm catalog mirror -a "${mirror_auths}" quay.io/openshift-qe-optional-operators/ocp4-index:latest ${MIRROR_REGISTRY_HOST} --manifests-only --to-manifests=/tmp/olm_mirror ; then
+    echo "$ cat /tmp/olm_mirror/imageContentSourcePolicy.yaml"
+    cat /tmp/olm_mirror/imageContentSourcePolicy.yaml
+    echo "$ oc create -f /tmp/olm_mirror/imageContentSourcePolicy.yaml"
+    if oc create -f /tmp/olm_mirror/imageContentSourcePolicy.yaml ; then
+        echo "create the ICSP resource successfully"
+    else
+        echo "!!!fail to create the ICSP resource"
+        echo "$ oc whoami"
+        oc whoami
+        exit 1
+    fi
+else
+    echo "!!!failed to generate the ICSP for OLM operators, list the mirror_auths:"
+    cat ${mirror_auths}
```

Comment:
please do not print credentials in log.

Reply:
I know, but I think it's not a best practice, or you can just print **keys** for debugging

Reply:
make sense.